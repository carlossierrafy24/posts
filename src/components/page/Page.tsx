import React from 'react';
import { User, Post, Comment } from '../../data/Interfaces';
import Card from '../card/Card';

type Props = {
    post: Post[];
    comment: Comment[];
    users: User[];
};

const Page = (props: Props) => {
    const { post, comment, users } = props;

    const selectCurrentUser = (id: number) => {
        const user = users.find((user) => user.id === id);
        return user;
    };

    const selectCurrentPost = (id: number) => {
        const cm = comment.filter((comment) => comment.postId === id);
        return cm;
    };

    return (
        <React.Fragment>
            <div className="grid-post-container-scroll mt-5 scroll-m-2 overflow-y-auto">
                <div className="grid-post-container">
                    {post.map((post) => (
                        <Card key={post.id} post={post} comments={selectCurrentPost(post.id)} user={selectCurrentUser(post.userId)} />
                    ))}
                </div>
            </div>
        </React.Fragment>
    );
};

export default Page;
