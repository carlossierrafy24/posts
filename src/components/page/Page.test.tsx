import React from 'react';
import { render } from '@testing-library/react';
import { Comment, Post, User } from '../../data/Interfaces';
import Page from './Page';

type Props = {
    post: Post[];
    comment: Comment[];
    users: User[];
};

const props: Props = {
    post: [
        {
            id: 1,
            title: 'Post title',
            body: 'Post body',
            userId: 1
        }
    ],
    comment: [
        {
            postId: 1,
            id: 1,
            name: 'Comment name',
            email: 'Comment email',
            body: 'Comment body'
        },
        {
            postId: 1,
            id: 1,
            name: 'Comment name',
            email: 'Comment email',
            body: 'Comment body'
        }
    ],
    users: [
        {
            id: 1,
            name: 'John Doe',
            username: 'johndoe',
            email: 'johndoe@gmail.com',
            address: {
                street: 'street',
                suite: 'place 2',
                city: 'city',
                zipcode: '000000',
                geo: {
                    lat: '000000000',
                    lng: '000000000'
                }
            },
            phone: '0000000000',
            website: 'john.doe.web',
            company: {
                name: 'loren ipsum',
                catchPhrase: 'lorem ipsum',
                bs: 'lorem ipsum'
            }
        }
    ]
};

test('should render page', () => {
    expect(render(<Page {...props} />)).toMatchSnapshot();
});
