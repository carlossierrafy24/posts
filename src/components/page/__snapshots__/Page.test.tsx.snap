// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid-post-container-scroll mt-5 scroll-m-2 overflow-y-auto"
      >
        <div
          class="grid-post-container"
        >
          <div
            class="post-container row-auto h-auto bg-blue-200"
          >
            <div
              class=""
            >
              <div
                class="flex items-center place-content-between"
              >
                <div
                  class="flex-row flex items-center p-2"
                >
                  <div
                    class="post-avatar ml-1 mr-3"
                  >
                    <img
                      alt="user_img"
                      class="rounded-full"
                      src="https://i.pravatar.cc/50"
                    />
                  </div>
                  <div
                    class=""
                  >
                    <p>
                      John Doe
                    </p>
                  </div>
                </div>
                <div
                  class="text-end pb-0"
                >
                  <button
                    class="comment-user-btn"
                  >
                    View 
                    2
                     comments
                  </button>
                </div>
              </div>
              <div
                class="text-lg text-center mb-3 font-bold"
              >
                <h2>
                  Post title
                </h2>
              </div>
              <div
                class="post-body-text italic px-10"
              >
                <p>
                  "
                  Post body
                  "
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid-post-container-scroll mt-5 scroll-m-2 overflow-y-auto"
    >
      <div
        class="grid-post-container"
      >
        <div
          class="post-container row-auto h-auto bg-blue-200"
        >
          <div
            class=""
          >
            <div
              class="flex items-center place-content-between"
            >
              <div
                class="flex-row flex items-center p-2"
              >
                <div
                  class="post-avatar ml-1 mr-3"
                >
                  <img
                    alt="user_img"
                    class="rounded-full"
                    src="https://i.pravatar.cc/50"
                  />
                </div>
                <div
                  class=""
                >
                  <p>
                    John Doe
                  </p>
                </div>
              </div>
              <div
                class="text-end pb-0"
              >
                <button
                  class="comment-user-btn"
                >
                  View 
                  2
                   comments
                </button>
              </div>
            </div>
            <div
              class="text-lg text-center mb-3 font-bold"
            >
              <h2>
                Post title
              </h2>
            </div>
            <div
              class="post-body-text italic px-10"
            >
              <p>
                "
                Post body
                "
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
